crates/tiny/config.yml

```diff
@@ -16,14 +16,20 @@ servers:
       # Three authentication methods: pass, sasl, and nickserv_ident
       # These are optional and you probably only need one of these, delete
       # others.
+      # For SASL EXTERNAL certificate and fingerprint generation, see server documentation.
+      # You will need to register the cert's fingerprint with NickServ
+      # ex. https://www.oftc.net/NickServ/CertFP/
 
       # Server or nick password
       # pass: 'hunter2'
 
       # SASL authentication
       # sasl:
-      #   username: 'tiny_user'
-      #   password: 'hunter2'
+      #   plain:
+      #     username: 'tiny_user'
+      #     password: 'hunter2'
+      #   external:
+      #     pem: "/home/.config/tiny/oftc.pem"
```

Comment:
This format will break user configs. Why not parse this to the right variant depending on the available keys? If `username` and `password` are the keys, then parse to `SASLAuth::Plain`. Otherwise parse to `SASLAuth::External`.

This will be more work (we can't derive `Deserialize` anymore), but we shouldn't avoid breaking user config to avoid a few minutes of work. I'm happy to help with implementing this if you (understandably) don't want to implement `Deserialize` for this type :-)

Reply:
Actually we can just do `#[serde(untagged)]` and that should resolve which variant is configured!